src/pages/api/recipients/[recId]/supply-requests.ts

```diff
@@ -79,8 +89,62 @@ async function handler(
       } catch (e) {
         return serverErrorHandler(e, res);
       }
+    case 'POST':
+      try {
+        const prisma = new PrismaClient();
+        const recipientAddSupply = await prisma.recipient.findUnique({
+          where: {
+            id: Number(recId),
+          },
+        });
+
+        if (!recipientAddSupply) {
+          return res.status(400).json({
+            error: true,
+            message: `No recipient found for id: ${recId}`,
+          });
+        }
+
+        if (!isValidRecipient) {
+          return res.status(400).json({
+            error: true,
+            message: `Please login as an authorized recipient user to access this resource`,
+          });
+        }
+        const newSupplyRequest = req.body as Prisma.SupplyRequestCreateInput;
+        const { items } = req.body;
+        const itemsId = [];
+
+        if (Array.isArray(items)) {
+          for (let i = 0; i < items.length; i += 1) {
```

Comment:
I think you can scrap this code and have the input as an array of objects with ID fields rather than an array of IDs. (i.e. `[{id: 123}, {id: 12312}]`)